### PR TITLE
feat: add support for selecting the organization using env var

### PR DESCRIPTION
**Pull request by i4ki**
The environment variable `TM_CLOUD_ORGANIZATION` can be used to select the organization where data will be sent to.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -56,11 +56,6 @@ jobs:
       - name: check cloud info
         run: ./bin/terramate experimental cloud info
 
-      - name: check cloud sync
-        run: ./bin/terramate run --log-level=debug --disable-check-git-remote --cloud-sync-deployment -- echo "Hello from Terramate CI"
-        env:
-          GITHUB_TOKEN: ${{ github.token }}
-
```
Comment (i4ki) [inline .github/workflows/ci.yml:63]: removing this for now because the repository is not trusted yet

File: cmd/terramate/cli/cloud.go [hunk 2]
```diff
@@ -78,12 +78,43 @@ func (c *cli) checkSyncDeployment() {
...
+	// at this point we know user is onboarded, ie has at least 1 organization.
+	orgs := c.cred().organizations()
+
+	useOrgName := os.Getenv("TM_CLOUD_ORGANIZATION")
+	if useOrgName != "" {
+		var useOrgUUID string
+		for _, org := range orgs {
+			if org.Name == useOrgName {
+				if org.Status != "active" {
+					fatal(errors.E("selected organization %s is not active", useOrgName))
```
Comment (mariux) [inline cmd/terramate/cli/cloud.go:90]:
```suggestion
					fatal(errors.E("You are not yet an active member of organization %s. Please accept the invitation first.", useOrgName))
```

File: cmd/terramate/cli/cloud.go [hunk 3]
```diff
@@ -78,12 +78,43 @@ func (c *cli) checkSyncDeployment() {
...
+				}
+
+				useOrgUUID = org.UUID
+				break
+			}
+		}
+
+		if useOrgUUID == "" {
+			fatal(errors.E("organization %q is not in the set of user's organizations: %s",
```
Comment (mariux) [inline cmd/terramate/cli/cloud.go:99]:
```suggestion
			fatal(errors.E("You are not a member of organization %q or the organization does not exist. Available organizations: %s",
```

File: cmd/terramate/cli/cloud.go [hunk 4]
```diff
@@ -78,12 +78,43 @@ func (c *cli) checkSyncDeployment() {
...
+
+		c.cloud.run.orgUUID = useOrgUUID
+	} else if len(orgs) != 1 {
 		fatal(
-			errors.E("requires 1 organization associated with the credential but %d found: %s",
-				len(orgs),
-				orgs),
+			errors.E("if the organization is not explicitly set with TM_CLOUD_ORGANIZATION "+
+				"environment variable then the user must be associated with only 1 organization "+
+				"but %d found: %s", len(orgs), orgs),
```
Comment (mariux) [inline cmd/terramate/cli/cloud.go:110]:
```suggestion
			errors.E("Please set TM_CLOUD_ORGANIZATION environment variable to a specific available organization %s", len(orgs), orgs),
```

File: cmd/terramate/cli/cloud.go [hunk 5]
```diff
@@ -78,12 +78,43 @@ func (c *cli) checkSyncDeployment() {
...
 		)
+	} else {
+		org := orgs[0]
+		if org.Status != "active" {
+			fatal(errors.E("selected organization %s is not active", org.Name))
```
Comment (mariux) [inline cmd/terramate/cli/cloud.go:115]: what is this case?

**Review by mariux — DISMISSED** [review]

File: cmd/terramate/cli/cloud.go (hunk 5 above)
Comment (i4ki) [reply cmd/terramate/cli/cloud.go:115]: if the user is a member of a single organization, then this org is automatically selected.

Comment (i4ki) [reply cmd/terramate/cli/cloud.go]: I improved the message but this `else` block is when there's a single organization associated with the user and no TM_CLOUD_ORGANIZATION is set, then it uses this one.
You commented on the fatal that aborts in the case that the member still didn't accept the invite for this organization. I improved the error message.

File: cmd/terramate/cli/cloud.go (hunk 4 above)
Comment (i4ki) [reply cmd/terramate/cli/cloud.go]: improved in a separate commit because the formatting is wrong

File: cmd/terramate/cli/cloud.go [hunk 6]
```diff
@@ -78,12 +78,41 @@ func (c *cli) checkSyncDeployment() {
...
+					fatal(errors.E("You are not yet an active member of organization %s. Please accept the invitation first.", useOrgName))
+				}
+
+				useOrgUUID = org.UUID
+				break
+			}
+		}
+
+		if useOrgUUID == "" {
+			fatal(errors.E("You are not a member of organization %q or the organization does not exist. Available organizations: %s",
```
Comment (MordecaiMalignatus) [inline cmd/terramate/cli/cloud.go:99]: Nitpick: Formatting an org list as a single string is going to be awkward to read and awkward to interact with, doesn't make for a great displayed message. Fine for now, not actually much of a problem, should be addressed if we make a big sweep of improving log/user messages.

File: cmd/terramate/cli/cloud.go [hunk 7]
```diff
@@ -78,12 +78,41 @@ func (c *cli) checkSyncDeployment() {
...
+			))
+		}
+
+		c.cloud.run.orgUUID = useOrgUUID
+	} else if len(orgs) != 1 {
 		fatal(
-			errors.E("requires 1 organization associated with the credential but %d found: %s",
-				len(orgs),
-				orgs),
+			errors.E("Please set TM_CLOUD_ORGANIZATION environment variable to a specific available organization %s", orgs),
```
Comment (MordecaiMalignatus) [inline cmd/terramate/cli/cloud.go:108]: Suggestion: We should consider in the future to make this configurable via files, and not just environment variables. As it is exceedingly unlikely for anyone to run into this right now, this is not actually a problem.

**Review by MordecaiMalignatus — APPROVED** [review]

**Review by MordecaiMalignatus — APPROVED** [review]

File: cmd/terramate/cli/cloud.go (hunk 7 above)
Comment (i4ki) [reply cmd/terramate/cli/cloud.go]: for sure it'd be great to maybe also support a command line flag.

**Pull request merged**